src/helpers/getOrigin.ts

```diff
@@ -0,0 +1,14 @@
+import getConfig from 'next/config';
+
+export const getOrigin = (): string | Error => {
+  const {
+    publicRuntimeConfig: { origin },
+  } = getConfig();
+
+  if (!origin) {
+    throw new Error(
+      'Please add a key "origin" in "publicRuntimeConfig" object in next.config.js file'
```

Comment:
This should specify that `origin` should contain the fully-qualified protocol + domain + path pointing to the root of the Next.js instance.